# <center>Projeto Final do Módulo VI</center>
<center> Alguns gastos suspeitos identificados

<center><img src='https://i.imgur.com/QLJaHpa.png' style="width:200px">

<center><a href="https://colab.research.google.com/github/Sartorio12/Projeto_Modulo_5/blob/main/notebook1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Grupo 9:** <a href="https://github.com/LeGuigs" target="_blank">Guilherme Rezende</a> ; <a href="https://github.com/LaioGomes" target="_blank">Laio Gomes</a> ; <a href="https://github.com/sartorio12" target="_blank">Mateus Sartorio</a>; <a href="https://github.com/jonathaslkc" target="_blank">Jonathas Carvalho</a>



<center><img src='https://i.imgur.com/QadqCNH.jpg' style="width:600px">
<center>Fonte: Agência Câmara de Notícias — Roque de Sá/Agência Senado

---

# **1.** Sobre o estudo e objetivos

O que você imagina quando ouve o conceito **"Cota Parlamentar"** ou **"Cota para o Exercício da Atividade Parlamentar (CEAP)"**?

Você sabia que é através da **CEAP** que deputados e senadores custeiam suas despesas em atividades parlamentares? O mais interessante disso, é que os dados da CEAP que iremos analisar hoje são abertos e podem ser acessados por qualquer pesso, ou seja, qualquer pessoa pode ver como os deputados e senadores utilizam essse tipo de recurso. 

Apesar de cobrir uma quantidade grande de despesas, os gastos com a CEAP devem seguir uma série de regras definidas pelo Ato da Mesa 43/2009. Isso quer dizer que os parlamentares não podem utilizar esses recursos de qualquer maneira. Alguns exemplos de gastos vedados são a aquisição de gêneros alimentícios (ex.: o parlamentar pode gastar com as suas refeições mas não pode comprar carne para o churrasco no fim de semana), gastos de caráter eleitoral e gastos com a participação em cursos de educação básica, graduação e pós-graduação.

Além disso, como citado anteriormente, a CEAP é destinada a custear gastos exclusivamente vinculados ao exercício da atividade parlamentar, o que veda o seu uso para o benefício de terceiros, como por exemplo, o pagamento de refeições para outras pessoas além do próprio parlamentar.
Apesar de todas essas regras, a fiscalização desses gastos é bastante falha, sendo comum o surgimento de denúncias sobre irregularidades.

Neste cenário surgiu um projeto chamado Operação Serenata de Amor ( https://serenata.ai/ ), que tem como objetivo utilizar ferramentas de ciência de dados para fiscalizar gastos públicos e compartilhar informações relevantes com o público. O projeto tem uma série de módulos. Alguns deles são responsáveis por baixar e limpar os dados, outros por identificar irregularidades e por fim existem alguns módulos responsáveis pela divulgação de eventuais alertas de irregularidades. Um desses módulos de divulgação é o bot @RosieDaSerenata, responsável por publicar alertas de irregularidades no Twitter.

<center><img src='https://i.imgur.com/AjyWlf1.jpg'>

**É sobre isso que vamos tratar nesta análise, que tem dois focos principais:**
- *Demonstrar como investigaremos esses dados utilizando Python.*
- *Elaborar conceitos a respeito da utilização da CEAP nos últimos 3 anos, com destaques no cenário de pandemia(COVID-19).*

#### <center> **Nós do grupo 5 esperamos que a leitura desse notebook seja esclarecedora e útil :)**

# **2.** Começando a nossa análise

Essa parte será um pouco teórica, pois nela exibiremos um pouco sobre a limpeza dos dados coletados no site da Câmara. Caso você queira somente acompanhar quais foram as nossas deduções baseadas nessas informações, pode pular para a parte X ( Preciso preencher a parte ).

*Antes de mais nada, vamos apresentar um glossário sobre o DataSet:*

- `Nome Parlamentar(txNomeParlamentar)` :  *Nome adotado pelo Parlamentar ao tomar posse do seu mandato.*
- `ideCadastro(Identificador Único do Parlamentar)` : *Número que identifica unicamente um deputado federal na CD.*
- `nuCarteiraParlamentar(Número da Carteira Parlamentar)` : *Documento usado para identificar um deputado federal na CD.*
- `nuLegislatura(Número da  Legislatura)` : *No contexto da cota CEAP, representa o ano base de início da legislatura.*
- `sgUF(Sigla da Unidade Federal)` : *Representa a UF do parlamentar.*
- `sdPartido(Sigla do Partido)` : *Representa o partido do parlamentar.*
- `codLegislatura(Código da Legislatura)` : *No contexto da cota CEAP, o seu conteúdo representa o código identificador da Legislatura.*
- `numSubCota(Número da Subcota) `: *Representa o código do Tipo de Despesa, comprovada por meio da emissão de um documento fiscal, a qual é debitada na cota do deputado.*
- `txtDescricao(Descrição da Subcota)` : *Descrição do Tipo de Despesa.*
- `numEspecificacaoSubCota(Número da Especificação da Subcota)` : *Representa o código desta especificação mais detalhada.*
- `txtDescricaoEspecificacao(Descrição da Especificação da Subcota)` : *Detalha o tipo de despesa efetuada.*
- `txtFornecedor(Fornecedor)` : *Representa o fornecedor do produto ou serviço presente no documento fiscal.*
- `txtCNPJCPF(CNPJ/CPF)` : *Indica se o fornecedor é CNPJ ou CPF , presente no documento fiscal.*
- `txtNumero(Número do Documento)` : *Indica o número do documento do emissor da NF em questão.*
- `indTipoDocumento(Indicativo de Tipo de Documento Fiscal)` : *Tipo de documento do fiscal – 0 para Nota Fiscal; 1 para Recibo; 2 para Despesa no Exterior.*
- `datEmissao(Data de Emissão)` : *A data de emissão do documento.*
- `vlrDocumento(Valor do Documento)` : *O seu conteúdo é o valor de face do documento fiscal ou o valor do documento que deu causa à despesa.*
- `vlrGlosa(Valor da Glosa)` : *Valor da glosa do documento que deu causa à despesa.*
- `vlrLiquido(Valor Liquido)` : *Valor líquido do documento (será calculado pela diferença entre o Valor do Documento e o Valor da Glosa)*
- `numMes(Mês) `: *Indica o mês da emissão da nota.*
- `numAno(Ano) `: *Indica o ano da emissão da nota.*
- `numParcela(Número da Parcela)` : *O seu conteúdo representa o número da parcela do documento fiscal. Ocorre quando o documento tem de ser reembolsado de forma parcelada.*
- `txtPassageiro(Nome do Passageiro)` : *O nome do passageiro, em caso da emissão do documento ser uma passagem aérea.*
- `txtTrecho(Trecho)` : *Representa qual o trecho da viagem, somente em casos de viagem aérea.*
- `numLote(Número do Lote) `: *Representa a capa do lote em que esse documento será incluído para ser entregue na Câmara para ressarcimento.*
- `numRessarcimento(Número do Ressarcimento) `: *Indica o ressarcimento do qual o documento faz parte.*
- `vlrRestituição(Valor da Restituição)` : *Indica o valor que será restituído sobre aquele documento.*
- `nuDeputadoId(Identificador do Solicitante)` : *Número que identifica um Parlamentar ou Liderança na Transparência da Cota para Exercício da Atividade Parlamentar.*

## **2.1.** Materiais e métodos

O presente trabalho foi desenvolvido utilizando `Google Colaboratory` e `Jupyter Notebook`, com a importação das seguintes bibliotecas:
> * `Pandas`: é uma biblioteca para uso em Python, open-source e de uso gratuito (sob uma licença BSD), que fornece ferramentas para análise e manipulação de dados. ;<br>
> * `Numpy`: é uma biblioteca para a linguagem Python com funções para se trabalhar com computação numérica. Seu principal objeto é o vetor n-dimensional, ou ndarray .;<br>
> * `Matplotlib`: é uma biblioteca com recursos para a geração de gráficos 2D a partir de arrays. Gráficos comuns podem ser criados com alta qualidade a partir de comandos simples, inspirados nos comandos gráficos do MATLAB;<br>
> * `Seaborn`: é uma biblioteca que atua em cima do matplotlib e ajuda a melhorar o visual dos gráficos, dando uma aparência mais bem acabada; e<br>
> * `Datetime`: é um módulo que fornece as classes para manipulação de datas e horas. Ainda que a aritmética de data e hora seja suportada, o foco da implementação é na extração eficiente do atributo para formatação da saída e manipulação.

A base de dados pode ser encontrada no site da Câmara ( https://www.camara.leg.br/cota-parlamentar/ ), de acesso livre.

Para versionamento do código utilizamos o `Git` e `GitHub` para publicação do trabalho.

Para organização e acompanhamento das tarefas entre os integrantes do grupo foi utilizado o `Trello`.

As reuniões foram estabelecidas através da plataforma `Discord`.

# **3.** Importando nossas bibliotecas

A parte incial, simples importação de bibliotecas que serão utilizadas.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## **3.2.** Importando nossos dados

In [2]:
df1 = pd.read_csv('Ano-2019.csv', sep=';')
df2 = pd.read_csv('Ano-2020.csv', sep=';')
df3 = pd.read_csv('Ano-2021.csv', sep=';')

## **3.3.** Começando com limpeza e verificação de dados

In [3]:
#Concatenando os 3 arquivos em um único
df_unico = pd.concat([df1,df2,df3])

In [4]:
# Verificando o número de partidos que contém o DataFrame
df_unico.sgPartido.value_counts()

PT               71008
PSL              67028
PP               44060
PL               38495
MDB              38487
PSD              36623
PSB              34907
PSDB             34371
REPUBLICANOS     29028
PDT              28706
DEM              27015
PTB              12719
SOLIDARIEDADE    12350
PODE             10574
CIDADANIA         9341
AVANTE            9257
PROS              8599
PSOL              8099
PCdoB             7288
PSC               6991
PATRIOTA          5951
NOVO              5829
PV                3761
PR                1118
REDE               708
PRB                531
S.PART.            377
PPS                225
PATRI              154
PHS                139
PPL                 20
Name: sgPartido, dtype: int64

In [5]:
# Criando uma cópia do dataframe para eventuais imprevistos que possam ocorrer
df_copia = df_unico.copy()

In [6]:
# Verificando o nome das colunas para facilitar na análise
df_copia.columns

Index(['txNomeParlamentar', 'cpf', 'ideCadastro', 'nuCarteiraParlamentar',
       'nuLegislatura', 'sgUF', 'sgPartido', 'codLegislatura', 'numSubCota',
       'txtDescricao', 'numEspecificacaoSubCota', 'txtDescricaoEspecificacao',
       'txtFornecedor', 'txtCNPJCPF', 'txtNumero', 'indTipoDocumento',
       'datEmissao', 'vlrDocumento', 'vlrGlosa', 'vlrLiquido', 'numMes',
       'numAno', 'numParcela', 'txtPassageiro', 'txtTrecho', 'numLote',
       'numRessarcimento', 'vlrRestituicao', 'nuDeputadoId', 'ideDocumento',
       'urlDocumento'],
      dtype='object')

In [7]:
# Conferindo o tipo de dado em cada coluna
df_copia.dtypes

txNomeParlamentar             object
cpf                          float64
ideCadastro                  float64
nuCarteiraParlamentar        float64
nuLegislatura                  int64
sgUF                          object
sgPartido                     object
codLegislatura                 int64
numSubCota                     int64
txtDescricao                  object
numEspecificacaoSubCota        int64
txtDescricaoEspecificacao     object
txtFornecedor                 object
txtCNPJCPF                    object
txtNumero                     object
indTipoDocumento               int64
datEmissao                    object
vlrDocumento                 float64
vlrGlosa                     float64
vlrLiquido                   float64
numMes                         int64
numAno                         int64
numParcela                     int64
txtPassageiro                 object
txtTrecho                     object
numLote                        int64
numRessarcimento             float64
v

In [8]:
# Vendo todas as informações e contando os valores não nulos em cada coluna
df_copia.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 555500 entries, 0 to 105604
Data columns (total 31 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   txNomeParlamentar          555500 non-null  object 
 1   cpf                        553661 non-null  float64
 2   ideCadastro                553759 non-null  float64
 3   nuCarteiraParlamentar      553759 non-null  float64
 4   nuLegislatura              555500 non-null  int64  
 5   sgUF                       553759 non-null  object 
 6   sgPartido                  553759 non-null  object 
 7   codLegislatura             555500 non-null  int64  
 8   numSubCota                 555500 non-null  int64  
 9   txtDescricao               555500 non-null  object 
 10  numEspecificacaoSubCota    555500 non-null  int64  
 11  txtDescricaoEspecificacao  136177 non-null  object 
 12  txtFornecedor              555467 non-null  object 
 13  txtCNPJCPF                 53

In [9]:
# Procurando numero de dados nulos em percentual
round(df_copia.isnull().mean()*100,2)

txNomeParlamentar             0.00
cpf                           0.33
ideCadastro                   0.31
nuCarteiraParlamentar         0.31
nuLegislatura                 0.00
sgUF                          0.31
sgPartido                     0.31
codLegislatura                0.00
numSubCota                    0.00
txtDescricao                  0.00
numEspecificacaoSubCota       0.00
txtDescricaoEspecificacao    75.49
txtFornecedor                 0.01
txtCNPJCPF                    4.28
txtNumero                     0.00
indTipoDocumento              0.00
datEmissao                    2.82
vlrDocumento                  0.00
vlrGlosa                      0.00
vlrLiquido                    0.00
numMes                        0.00
numAno                        0.00
numParcela                    0.00
txtPassageiro                76.69
txtTrecho                    76.88
numLote                       0.00
numRessarcimento             67.45
vlrRestituicao               99.93
nuDeputadoId        

In [10]:
# o describe foi feito pra destacar os valores negativos presentes nas colunas:'vlrDocumento' e 'vlrLiquido' 
df_copia.describe()

cpf    ideCadastro  nuCarteiraParlamentar  nuLegislatura  \
count  5.536610e+05  553759.000000          553759.000000  555500.000000   
mean   3.208493e+10  166015.342644             289.007700    2018.894387   
std    2.921625e+10   45864.510704             153.129661       0.641326   
min    1.608657e+06    4930.000000               1.000000    2015.000000   
25%    6.197278e+09  141531.000000             160.000000    2019.000000   
50%    2.227515e+10  178938.000000             292.000000    2019.000000   
75%    5.098035e+10  204449.000000             424.000000    2019.000000   
max    9.989866e+10  217330.000000             627.000000    2019.000000   

       codLegislatura     numSubCota  numEspecificacaoSubCota  \
count   555500.000000  555500.000000            555500.000000   
mean        55.973597     250.656994                 0.251131   
std          0.160331     412.069770                 0.457949   
min         55.000000       1.000000                 0.000000   
25%         56.000000       3.000000                 0.000000   
50%         56.000000      10.000000                 0.000000   
75%         56.000000     122.000000                 0.000000   
max         56.000000     999.000000                 4.000000   

       indTipoDocumento   vlrDocumento       vlrGlosa     vlrLiquido  \
count     555500.000000  555500.000000  555500.000000  555500.000000   
mean           1.029525     884.412172       3.900387     808.744693   
std            1.567946    2464.526292     139.446699    2464.253989   
min            0.000000   -6605.570000       0.000000   -7821.200000   
25%            0.000000      63.000000       0.000000      49.510000   
50%            0.000000     201.380000       0.000000     184.500000   
75%            1.000000     747.600000       0.000000     642.000000   
max            4.000000  172000.000000   86000.000000  151260.000000   

              numMes         numAno     numParcela       numLote  \
count  555500.000000  555500.000000  555500.000000  5.555000e+05   
mean        6.083879    2019.669870       0.000238  1.130421e+06   
std         3.337029       0.775475       0.015413  7.872364e+05   
min         1.000000    2019.000000       0.000000  0.000000e+00   
25%         3.000000    2019.000000       0.000000  0.000000e+00   
50%         6.000000    2019.000000       0.000000  1.620008e+06   
75%         9.000000    2020.000000       0.000000  1.707915e+06   
max        12.000000    2021.000000       1.000000  1.789366e+06   

       numRessarcimento  vlrRestituicao   nuDeputadoId  ideDocumento  
count          180842.0      364.000000  555500.000000  5.555000e+05  
mean                0.0      873.633571    2734.361870  5.045073e+06  
std                 0.0     2367.571616     773.342999  2.865350e+06  
min                 0.0        1.000000      19.000000  0.000000e+00  
25%                 0.0        6.975000    2265.000000  1.705442e+06  
50%                 0.0       26.105000    3062.000000  6.888214e+06  
75%                 0.0      277.395000    3312.000000  7.075930e+06  
max                 0.0    20000.000000    3487.000000  7.245337e+06

In [11]:
# Essa consulta fizemos um novo df, filtrando na coluna 'txtDescricao' valores diferentes de 'PASSAGEM AÉREA - RPA'  e 'PASSAGEM AÉREA - SIGEPA'
# a partir desse novo df fazer os cálculos de despesas.
valores = df_copia[(df_copia['txtDescricao'] != 'PASSAGEM AÉREA - RPA')&(df_copia['txtDescricao'] != 'PASSAGEM AÉREA - SIGEPA')]
valores['txtDescricao'].unique()
valores.vlrLiquido

0          111.30
1          117.19
2          120.15
3          151.18
4          119.90
           ...   
105600    5000.00
105601    7000.00
105602    4000.00
105603       1.10
105604       1.08
Name: vlrLiquido, Length: 427074, dtype: float64

In [12]:
# os dois valores negativos da coluna 'VrlLiquido'
valores.query('vlrLiquido < 0')['vlrLiquido']

23210    -868.00
28211   -1762.66
Name: vlrLiquido, dtype: float64

In [13]:
# Convertemos a coluna datEmissao de object para datatime e colocamos só pra mostrar o formato dia/mes/ano pq o horário é irrelevante para as consultas
df_copia['datEmissao'] = pd.to_datetime(df_copia['datEmissao'], format = '%Y/%m/%d')
df_copia['datEmissao']

0        2019-01-16
1        2019-02-04
2        2019-02-25
3        2019-03-28
4        2019-05-07
            ...    
105600   2021-07-28
105601   2021-08-11
105602   2021-07-29
105603          NaT
105604          NaT
Name: datEmissao, Length: 555500, dtype: datetime64[ns]

In [14]:
#Conferindo se o type da data foi transformado
df_copia.dtypes

txNomeParlamentar                    object
cpf                                 float64
ideCadastro                         float64
nuCarteiraParlamentar               float64
nuLegislatura                         int64
sgUF                                 object
sgPartido                            object
codLegislatura                        int64
numSubCota                            int64
txtDescricao                         object
numEspecificacaoSubCota               int64
txtDescricaoEspecificacao            object
txtFornecedor                        object
txtCNPJCPF                           object
txtNumero                            object
indTipoDocumento                      int64
datEmissao                   datetime64[ns]
vlrDocumento                        float64
vlrGlosa                            float64
vlrLiquido                          float64
numMes                                int64
numAno                                int64
numParcela                      

In [15]:
# Verificando o novo tamanho do DataFrame
df_copia.shape

(555500, 31)

In [16]:
# Aqui estamos procurando por Parlamentares sem estado e sem partido.
sem_partido_sem_uf = valores[valores['sgUF'].isnull()][['txNomeParlamentar', 'sgUF', 'sgPartido','vlrLiquido']]
sem_partido_sem_uf

txNomeParlamentar sgUF sgPartido  vlrLiquido
0          LID.GOV-CD  NaN       NaN      111.30
1          LID.GOV-CD  NaN       NaN      117.19
2          LID.GOV-CD  NaN       NaN      120.15
3          LID.GOV-CD  NaN       NaN      151.18
4          LID.GOV-CD  NaN       NaN      119.90
..                ...  ...       ...         ...
439            LIDMIN  NaN       NaN     4000.00
440            LIDMIN  NaN       NaN      137.97
441            LIDMIN  NaN       NaN      175.18
442            LIDMIN  NaN       NaN     3800.00
443            LIDMIN  NaN       NaN     3800.00

[1741 rows x 4 columns]

In [17]:
#Atribuindo os partidos correspondentes as despesas atribuídas apenas a lideranças
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO PSL", 'PSL', df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO PT", 'PT', df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO PTB", 'PTB', df_copia.sgPartido) 
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO PSDB", 'PSDB', df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO PODEMOS", 'PODEMOS', df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO SOLIDARIEDADE", 'SOLIDARIEDADE', df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO PSD", 'PSD',df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO NOVO", 'NOVO',df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO DEMOCRATAS", 'DEMOCRATAS', df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO CIDADANIA", 'CIDADANIA', df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO AVANTE", 'AVANTE', df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO PDT", 'PDT', df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO PSOL", 'PSOL', df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO PROGRESSISTAS", 'PROGRESSISTAS',df_copia.sgPartido)
df_copia['sgPartido'] = np.where(df_copia['txNomeParlamentar'] == "LIDERANÇA DO PARTIDO REPUBLICANO DA ORDEM SOCIAL", 'SOCIAL', df_copia.sgPartido)

In [18]:
# Verificando quantos partidos não possuem registros de UF
sem_partido_sem_uf['txNomeParlamentar'].value_counts()

LIDERANÇA DO PT                                     515
LIDERANÇA DO PTB                                    169
LIDERANÇA DO PSDB                                   168
LIDERANÇA DO PODEMOS                                140
LIDERANÇA DO SOLIDARIEDADE                          137
LIDERANÇA DO PSD                                    104
LIDERANÇA DO NOVO                                    94
LIDERANÇA DO DEMOCRATAS                              75
LIDERANÇA DO CIDADANIA                               68
LIDMIN                                               67
LIDERANÇA DO PARTIDO REPUBLICANO DA ORDEM SOCIAL     51
LIDERANÇA DO AVANTE                                  49
LIDERANÇA DO PDT                                     36
LIDERANÇA DO PSL                                     29
LID.GOV-CD                                           15
LIDERANÇA DO PSOL                                    14
LIDERANÇA DO PROGRESSISTAS                           10
Name: txNomeParlamentar, dtype: int64

In [19]:
# Realizando a contagem de frequência por estado dentro do DataFrame
df_unico.sgUF.value_counts()

SP    83873
MG    65193
RJ    50674
RS    48467
BA    43331
PR    41816
PE    28292
SC    19900
CE    19632
MA    14412
PA    14402
PB    12727
GO    11315
RN    10183
ES     9516
TO     9378
MT     9260
MS     7970
RO     7512
PI     7284
AC     6702
AL     6629
SE     6124
AM     5723
RR     5405
AP     5216
DF     2823
Name: sgUF, dtype: int64

In [20]:
#Verificando o funcionamento da atribuição
df_copia[df_copia['txNomeParlamentar'].str.match('LIDERANÇA')].sample(5)

txNomeParlamentar  cpf  ideCadastro  nuCarteiraParlamentar  \
765         LIDERANÇA DO PTB  NaN          NaN                    NaN   
278         LIDERANÇA DO PSD  NaN          NaN                    NaN   
610          LIDERANÇA DO PT  NaN          NaN                    NaN   
155     LIDERANÇA DO PODEMOS  NaN          NaN                    NaN   
42   LIDERANÇA DO DEMOCRATAS  NaN          NaN                    NaN   

     nuLegislatura sgUF   sgPartido  codLegislatura  numSubCota  \
765           2019  NaN         PTB              56          13   
278           2019  NaN         PSD              56          13   
610           2019  NaN          PT              56          13   
155           2019  NaN     PODEMOS              56          13   
42            2019  NaN  DEMOCRATAS              56          13   

                                   txtDescricao  ...  numAno numParcela  \
765  FORNECIMENTO DE ALIMENTAÇÃO DO PARLAMENTAR  ...    2019          0   
278  FORNECIMENTO DE ALIMENTAÇÃO DO PARLAMENTAR  ...    2019          0   
610  FORNECIMENTO DE ALIMENTAÇÃO DO PARLAMENTAR  ...    2019          0   
155  FORNECIMENTO DE ALIMENTAÇÃO DO PARLAMENTAR  ...    2021          0   
42   FORNECIMENTO DE ALIMENTAÇÃO DO PARLAMENTAR  ...    2021          0   

    txtPassageiro txtTrecho  numLote  numRessarcimento vlrRestituicao  \
765           NaN       NaN  1662683               NaN            NaN   
278           NaN       NaN  1590961               NaN            NaN   
610           NaN       NaN  1606679               NaN            NaN   
155           NaN       NaN  1747113               NaN            NaN   
42            NaN       NaN  1771915               NaN            NaN   

     nuDeputadoId  ideDocumento  \
765          2722       6977657   
278          2725       6821628   
610          2439       6856715   
155          3172       7157954   
42           2713       7208499   

                                          urlDocumento  
765  http://camara.leg.br/cota-parlamentar/nota-fis...  
278  https://www.camara.leg.br/cota-parlamentar/doc...  
610  http://camara.leg.br/cota-parlamentar/nota-fis...  
155  http://camara.leg.br/cota-parlamentar/nota-fis...  
42   http://camara.leg.br/cota-parlamentar/nota-fis...  

[5 rows x 31 columns]

In [21]:
#Criando uma cópia para trabalharmos nas questões e realizar as análises devidas
df_2 = df_copia.copy()

# MÓDULO 6 - TOCANTINS

In [22]:
#Mostrar as colunas para explorarmos o df
df_copia.columns

Index(['txNomeParlamentar', 'cpf', 'ideCadastro', 'nuCarteiraParlamentar',
       'nuLegislatura', 'sgUF', 'sgPartido', 'codLegislatura', 'numSubCota',
       'txtDescricao', 'numEspecificacaoSubCota', 'txtDescricaoEspecificacao',
       'txtFornecedor', 'txtCNPJCPF', 'txtNumero', 'indTipoDocumento',
       'datEmissao', 'vlrDocumento', 'vlrGlosa', 'vlrLiquido', 'numMes',
       'numAno', 'numParcela', 'txtPassageiro', 'txtTrecho', 'numLote',
       'numRessarcimento', 'vlrRestituicao', 'nuDeputadoId', 'ideDocumento',
       'urlDocumento'],
      dtype='object')

In [23]:
#Novo dataframe com o filtro só para deputados do estado de Tocantins
df_TO = df_copia[df_copia['sgUF'] == 'TO']

In [24]:
#Simples visualização dos primeiros registros
df_TO.head(2)

txNomeParlamentar           cpf  ideCadastro  nuCarteiraParlamentar  \
22968     FREIRE JÚNIOR  7.402202e+10      74189.0                  615.0   
22969     FREIRE JÚNIOR  7.402202e+10      74189.0                  615.0   

       nuLegislatura sgUF sgPartido  codLegislatura  numSubCota  \
22968           2015   TO       MDB              55           3   
22969           2015   TO       MDB              55           3   

                        txtDescricao  ...  numAno numParcela txtPassageiro  \
22968  COMBUSTÍVEIS E LUBRIFICANTES.  ...    2019          0           NaN   
22969  COMBUSTÍVEIS E LUBRIFICANTES.  ...    2019          0           NaN   

      txtTrecho  numLote  numRessarcimento vlrRestituicao  nuDeputadoId  \
22968       NaN  1552679               NaN            NaN           496   
22969       NaN  1557161               NaN            NaN           496   

       ideDocumento                                       urlDocumento  
22968       6733941  http://camara.leg.br/cota-parlamentar/nota-fis...  
22969       6744495  http://camara.leg.br/cota-parlamentar/nota-fis...  

[2 rows x 31 columns]

In [25]:
df_TO.isna().sum()

txNomeParlamentar               0
cpf                             0
ideCadastro                     0
nuCarteiraParlamentar           0
nuLegislatura                   0
sgUF                            0
sgPartido                       0
codLegislatura                  0
numSubCota                      0
txtDescricao                    0
numEspecificacaoSubCota         0
txtDescricaoEspecificacao    6440
txtFornecedor                   0
txtCNPJCPF                    474
txtNumero                       0
indTipoDocumento                0
datEmissao                    247
vlrDocumento                    0
vlrGlosa                        0
vlrLiquido                      0
numMes                          0
numAno                          0
numParcela                      0
txtPassageiro                7648
txtTrecho                    7649
numLote                         0
numRessarcimento             6614
vlrRestituicao               9371
nuDeputadoId                    0
ideDocumento  

In [26]:

df_tocantins = df_TO.sort_values(by=['vlrLiquido'], ascending=False)
df_tocantins.head()

txNomeParlamentar           cpf  ideCadastro  \
109715  Professora Dorinha Seabra Rezende  4.319693e+10     160639.0   
55884   Professora Dorinha Seabra Rezende  4.319693e+10     160639.0   
55885   Professora Dorinha Seabra Rezende  4.319693e+10     160639.0   
184326                        Célio Moura  1.236498e+10     204370.0   
30223                   Vicentinho Júnior  9.989866e+10     137070.0   

        nuCarteiraParlamentar  nuLegislatura sgUF sgPartido  codLegislatura  \
109715                   67.0           2019   TO       DEM              56   
55884                    67.0           2019   TO       DEM              56   
55885                    67.0           2019   TO       DEM              56   
184326                   62.0           2019   TO        PT              56   
30223                    66.0           2019   TO        PL              56   

        numSubCota                                   txtDescricao  ...  \
109715           5           DIVULGAÇÃO DA ATIVIDADE PARLAMENTAR.  ...   
55884            5           DIVULGAÇÃO DA ATIVIDADE PARLAMENTAR.  ...   
55885            5           DIVULGAÇÃO DA ATIVIDADE PARLAMENTAR.  ...   
184326           5           DIVULGAÇÃO DA ATIVIDADE PARLAMENTAR.  ...   
30223            4  CONSULTORIAS, PESQUISAS E TRABALHOS TÉCNICOS.  ...   

        numAno numParcela txtPassageiro txtTrecho  numLote  numRessarcimento  \
109715    2019          0           NaN       NaN  1661126               NaN   
55884     2020          0           NaN       NaN  1733387               NaN   
55885     2020          0           NaN       NaN  1728438               NaN   
184326    2019          0           NaN       NaN  1665905               NaN   
30223     2020          0           NaN       NaN  1668289               NaN   

       vlrRestituicao  nuDeputadoId  ideDocumento  \
109715            NaN          2300       6981838   
55884             NaN          2300       7129412   
55885             NaN          2300       7119201   
184326            NaN          3203       6991464   
30223             NaN          3059       6997536   

                                             urlDocumento  
109715  https://www.camara.leg.br/cota-parlamentar/doc...  
55884   https://www.camara.leg.br/cota-parlamentar/doc...  
55885   https://www.camara.leg.br/cota-parlamentar/doc...  
184326  https://www.camara.leg.br/cota-parlamentar/doc...  
30223   http://camara.leg.br/cota-parlamentar/nota-fis...  

[5 rows x 31 columns]

In [27]:
# Professora Dorinha gasta pra caramba

dfteste = df_TO[df_TO['txNomeParlamentar'].str.contains('Osires Damaso')]
dfteste2 = dfteste.sort_values(by=['vlrLiquido'], ascending=False)
dfteste2.head()

txNomeParlamentar           cpf  ideCadastro  nuCarteiraParlamentar  \
93687      Osires Damaso  2.784828e+10     204363.0                   64.0   
93688      Osires Damaso  2.784828e+10     204363.0                   64.0   
181062     Osires Damaso  2.784828e+10     204363.0                   64.0   
93680      Osires Damaso  2.784828e+10     204363.0                   64.0   
62624      Osires Damaso  2.784828e+10     204363.0                   64.0   

        nuLegislatura sgUF sgPartido  codLegislatura  numSubCota  \
93687            2019   TO       PSC              56           4   
93688            2019   TO       PSC              56           4   
181062           2019   TO       PSC              56           4   
93680            2019   TO       PSC              56           4   
62624            2019   TO       PSC              56           4   

                                         txtDescricao  ...  numAno numParcela  \
93687   CONSULTORIAS, PESQUISAS E TRABALHOS TÉCNICOS.  ...    2020          0   
93688   CONSULTORIAS, PESQUISAS E TRABALHOS TÉCNICOS.  ...    2020          0   
181062  CONSULTORIAS, PESQUISAS E TRABALHOS TÉCNICOS.  ...    2019          0   
93680   CONSULTORIAS, PESQUISAS E TRABALHOS TÉCNICOS.  ...    2020          0   
62624   CONSULTORIAS, PESQUISAS E TRABALHOS TÉCNICOS.  ...    2021          0   

       txtPassageiro txtTrecho  numLote  numRessarcimento vlrRestituicao  \
93687            NaN       NaN  1735664               NaN            NaN   
93688            NaN       NaN  1736650               NaN            NaN   
181062           NaN       NaN  1617492               NaN            NaN   
93680            NaN       NaN  1705163               NaN            NaN   
62624            NaN       NaN  1743009               NaN            NaN   

        nuDeputadoId  ideDocumento  \
93687           3239       7133923   
93688           3239       7136260   
181062          3239       6882641   
93680           3239       7070520   
62624           3239       7149573   

                                             urlDocumento  
93687   https://www.camara.leg.br/cota-parlamentar/doc...  
93688   https://www.camara.leg.br/cota-parlamentar/doc...  
181062  http://camara.leg.br/cota-parlamentar/nota-fis...  
93680   http://camara.leg.br/cota-parlamentar/nota-fis...  
62624   http://camara.leg.br/cota-parlamentar/nota-fis...  

[5 rows x 31 columns]

tiago dimas suspeito

# Cnpj's

In [28]:
#Instalação do package requests para capturarmos dados por meio da url
# !pip3 install requests

In [29]:
# #Importação das libs que trabalharemos e configurar display para visualizar todas as colunas (jupyter)
# import requests
# import json as js
# pd.set_option('display.max_columns', None)

In [30]:
# #Seleção dos cnpj/cpf com total de soma e ordenado para somente os 10 primeiros. Transformandos a series em df e aplicamos...
# #nome da coluna de retorno do groupby de Valor e resetamos o index
# df_gasto = pd.DataFrame({'Valor': df_copia.groupby(['txtCNPJCPF'])['vlrDocumento'].sum().sort_values(ascending = False).head(10)}).reset_index()
# df_gasto

In [31]:
#Atribuímos o retorno do get no formado json à nova coluna dados_CNPJ em nosso dataframe do groupby
# df_gasto['dados_CNPJ'] = df_gasto['txtCNPJCPF'].map(lambda x: requests.get(f"https://minhareceita.org/{x}"))

In [32]:
#Atribuímos a formatação do json para modelo de dicionário
# df_gasto['dados_CNPJ'] = df_gasto['dados_CNPJ'].map(lambda x: js.loads(x.content))

In [33]:
#Simples visualização formatando o dict em list e convertendo-o em dataframe
# pd.DataFrame(list(df_gasto['dados_CNPJ']))

# Laio

# Sartorio

In [34]:
dfteste3 = df_TO['txNomeParlamentar'].value_counts()
dfteste3['vlrGlosa']

KeyError: 'vlrGlosa'

In [44]:
# Professora Dorinha gasta pra caramba

dfteste = df_TO[df_TO['txtDescricao'].str.contains('POSTA')]
dfteste2 = dfteste.sort_values(by=['vlrDocumento'], ascending=False)
dfteste2

txNomeParlamentar           cpf  ideCadastro  \
56037   Professora Dorinha Seabra Rezende  4.319693e+10     160639.0   
56040   Professora Dorinha Seabra Rezende  4.319693e+10     160639.0   
78798                         Tiago Dimas  1.657051e+09     143084.0   
36484   Professora Dorinha Seabra Rezende  4.319693e+10     160639.0   
57059                       Dulce Miranda  3.021787e+10     178994.0   
...                                   ...           ...          ...   
169031                      Dulce Miranda  3.021787e+10     178994.0   
169038                      Dulce Miranda  3.021787e+10     178994.0   
169041                      Dulce Miranda  3.021787e+10     178994.0   
59128                   Vicentinho Júnior  9.989866e+10     137070.0   
169057                      Dulce Miranda  3.021787e+10     178994.0   

        nuCarteiraParlamentar  nuLegislatura sgUF      sgPartido  \
56037                    67.0           2019   TO            DEM   
56040                    67.0           2019   TO            DEM   
78798                    65.0           2019   TO  SOLIDARIEDADE   
36484                    67.0           2019   TO            DEM   
57059                    61.0           2019   TO            MDB   
...                       ...            ...  ...            ...   
169031                   61.0           2019   TO            MDB   
169038                   61.0           2019   TO            MDB   
169041                   61.0           2019   TO            MDB   
59128                    66.0           2019   TO             PL   
169057                   61.0           2019   TO            MDB   

        codLegislatura  numSubCota      txtDescricao  ...  numAno numParcela  \
56037               56          11  SERVIÇOS POSTAIS  ...    2020          0   
56040               56          11  SERVIÇOS POSTAIS  ...    2020          0   
78798               56          11  SERVIÇOS POSTAIS  ...    2019          0   
36484               56          11  SERVIÇOS POSTAIS  ...    2021          0   
57059               56          11  SERVIÇOS POSTAIS  ...    2021          0   
...                ...         ...               ...  ...     ...        ...   
169031              56          11  SERVIÇOS POSTAIS  ...    2019          0   
169038              56          11  SERVIÇOS POSTAIS  ...    2019          0   
169041              56          11  SERVIÇOS POSTAIS  ...    2019          0   
59128               56          11  SERVIÇOS POSTAIS  ...    2019          0   
169057              56          11  SERVIÇOS POSTAIS  ...    2019          0   

       txtPassageiro txtTrecho  numLote  numRessarcimento vlrRestituicao  \
56037            NaN       NaN        0               0.0            NaN   
56040            NaN       NaN        0               0.0            NaN   
78798            NaN       NaN  1642835               NaN            NaN   
36484            NaN       NaN        0               0.0            NaN   
57059            NaN       NaN        0               0.0            NaN   
...              ...       ...      ...               ...            ...   
169031           NaN       NaN        0               0.0            NaN   
169038           NaN       NaN        0               0.0            NaN   
169041           NaN       NaN        0               0.0            NaN   
59128            NaN       NaN        0               0.0            NaN   
169057           NaN       NaN        0               0.0            NaN   

        nuDeputadoId  ideDocumento  \
56037           2300             0   
56040           2300             0   
78798           3296       6940055   
36484           2300             0   
57059           2905             0   
...              ...           ...   
169031          2905             0   
169038          2905             0   
169041          2905             0   
59128           3059             0   
169057          2905             0   

           

In [ ]:
df_TO.sort_values(by=['vlrGlosa'], ascending=False)

In [45]:
dfteste2['vlrDocumento'].mean()

156.37123543123542

In [ ]:
dfteste2.query('vlrLiquido > 60')

Optamos por escolher alguns critérios que levantariam suspeitas em gastos parlamentares, seriam esses:

CNPJ inválido
Valores discrepantes para o gasto em questão
Extrapolar o limite da subquota ( TO 39.503,61 )

Encontrei alguns valores um pouco absurdos para uma refeição, pois seja almoço, jantar ou uma simples merenda, a legislação que trata da CEAP diz que o reembolso referente à alimentação somente será autorizado para despesas com o próprio parlamentar (nem assessor entra).

Há valores absurdos no conjunto de dados de reembolsos no tocante à subquota parlamentar destinada à refeição dos parlamentares.

Em relação a alimentação, a média geral é de 40 reais por refeição, valores muito acima disso são considerados suspeitos também.

Número de parlamentares e número de gastos lançados no período em questão:

Célio Moura                          1855
Tiago Dimas                          1589
Eli Borges                           1319
Professora Dorinha Seabra Rezende    1243
Dulce Miranda                        1098
Carlos Henrique Gaguim               1057
Osires Damaso                         802
Vicentinho Júnior                     381
LÁZARO BOTELHO                         12
JOSI NUNES                             11
FREIRE JÚNIOR                           6
CÉSAR HALUM                             5

Tiago Dimas, do partido Solidariedade, gastou no COCO BAMBU LAGO SUL o valor de 206,90 em um único dia ( 11/07/2019 ), em uma única alimentação, segundo a nota fiscal enviada, o gasto foi utilizado de forma indevida, pois foram pedidos diversos itens, entre eles 

Célio Moura, do PT, gastou 185 reais no dia 12/10/2020 no restaurante SANTO CANTO 




In [ ]:
dfgastos3 = df_TO.query('vlrLiquido > 0')
dfgastos3['txNomeParlamentar'].value_counts()

Célio Moura                          1800
Tiago Dimas                          1393
Eli Borges                           1312
Professora Dorinha Seabra Rezende    1076
Dulce Miranda                        1020
Carlos Henrique Gaguim               1008
Osires Damaso                         755
Vicentinho Júnior                     335
JOSI NUNES                             11
LÁZARO BOTELHO                         11
FREIRE JÚNIOR                           6
CÉSAR HALUM                             5
Name: txNomeParlamentar, dtype: int64

In [ ]:
df_TO

txNomeParlamentar           cpf  ideCadastro  nuCarteiraParlamentar  \
22968     FREIRE JÚNIOR  7.402202e+10      74189.0                  615.0   
22969     FREIRE JÚNIOR  7.402202e+10      74189.0                  615.0   
22970     FREIRE JÚNIOR  7.402202e+10      74189.0                  615.0   
22971     FREIRE JÚNIOR  7.402202e+10      74189.0                  615.0   
22972     FREIRE JÚNIOR  7.402202e+10      74189.0                  615.0   
...                 ...           ...          ...                    ...   
64290       Célio Moura  1.236498e+10     204370.0                   62.0   
64291       Célio Moura  1.236498e+10     204370.0                   62.0   
64292       Célio Moura  1.236498e+10     204370.0                   62.0   
64293       Célio Moura  1.236498e+10     204370.0                   62.0   
64294       Célio Moura  1.236498e+10     204370.0                   62.0   

       nuLegislatura sgUF sgPartido  codLegislatura  numSubCota  \
22968           2015   TO       MDB              55           3   
22969           2015   TO       MDB              55           3   
22970           2015   TO       MDB              55           3   
22971           2015   TO       MDB              55           4   
22972           2015   TO       MDB              55           9   
...              ...  ...       ...             ...         ...   
64290           2019   TO        PT              56         122   
64291           2019   TO        PT              56         122   
64292           2019   TO        PT              56         122   
64293           2019   TO        PT              56         122   
64294           2019   TO        PT              56         998   

                                        txtDescricao  numEspecificacaoSubCota  \
22968                  COMBUSTÍVEIS E LUBRIFICANTES.                        1   
22969                  COMBUSTÍVEIS E LUBRIFICANTES.                        1   
22970                  COMBUSTÍVEIS E LUBRIFICANTES.                        1   
22971  CONSULTORIAS, PESQUISAS E TRABALHOS TÉCNICOS.                        0   
22972                     PASSAGEM AÉREA - REEMBOLSO                        0   
...                                              ...                      ...   
64290      SERVIÇO DE TÁXI, PEDÁGIO E ESTACIONAMENTO                        0   
64291      SERVIÇO DE TÁXI, PEDÁGIO E ESTACIONAMENTO                        0   
64292      SERVIÇO DE TÁXI, PEDÁGIO E ESTACIONAMENTO                        0   
64293      SERVIÇO DE TÁXI, PEDÁGIO E ESTACIONAMENTO                        0   
64294                        PASSAGEM AÉREA - SIGEPA                        0   

      txtDescricaoEspecificacao  \
22968      Veículos Automotores   
22969      Veículos Automotores   
22970      Veículos Automotores   
22971                       NaN   
22972                       NaN   
...                         ...   
64290                       NaN   
64291                       NaN   
64292                       NaN   
64293                       NaN   
64294                       NaN   

                                           txtFornecedor          txtCNPJCPF  \
22968  AUTO SHOPPING QL 06 COM. DE DER. DE PETROLEO LTDA  006.474.400/0013-5   
22969  AUTO SHOPPING QL 06 COM. DE DER. DE PETROLEO LTDA  006.474.400/0013-5   
22970          CRUZ & CRUZ COMERCIO DE COMBUSTIVEIS LTDA  040.064.710/0016-8   
22971           RACHID SOCIEDADE INDIVIDUAL DE ADVOCACIA  172.705.550/0015-6   
22972                              GOL LINHAS AÉREAS S.A  075.756.510/0015-9   
...                                                  ...                 ...   
64290                    UBER DO BRASIL TECNOLOGIA LTDA.  178.956.460/0018-7   
64291                    UBER DO BRASIL TECNOLOGIA LTDA.  178.956.460/0018-7   
64292                    UBER DO BRASIL TECNOLOGIA LTDA.  178.956.460/0018-7   
64293                UBER DO BRASIL TECNOLOGIA LTDA.  **  178.956.460/0018-7   
64

In [ ]:
# O Deputado tiago dimas tem muitos gastos em valores exatos declarados como CONSULTORIAS, PESQUISAS E TRABALHOS TÉCNICOS.

# Gui

# Jonathas